be/src/formats/csv/converter.cpp

```diff
@@ -75,4 +75,60 @@ std::unique_ptr<Converter> get_converter(const TypeDescriptor& type_desc, bool n
     return nullable ? std::make_unique<NullableConverter>(std::move(c)) : std::move(c);
 }
 
+// Hive collection delimiter generate rule refer to:
+// https://github.com/apache/hive/blob/master/serde/src/java/org/apache/hadoop/hive/serde2/lazy/LazySerDeParameters.java#L250
+static char get_collection_delimiter(char mapkey_delimiter, size_t nested_array_level) {
+    DCHECK(nested_array_level > 1 && nested_array_level < 153);
+
+    if (nested_array_level == 2) {
+        return mapkey_delimiter;
+    }
+
+    // tmp maybe negative, dont use size_t.
+    int32_t tmp;
+    if (nested_array_level <= 7) {
```

Comment:
plz add comments

Reply:
I add detailed comments in `array_converter.cpp`.